syncmaster/backend/services/get_user.py

```diff
@@ -0,0 +1,50 @@
+# SPDX-FileCopyrightText: 2023-2024 MTS PJSC
+# SPDX-License-Identifier: Apache-2.0
+from collections.abc import Callable, Coroutine
+from typing import Annotated, Any
+
+from fastapi import Depends, Request, status
+from fastapi.exceptions import HTTPException
+from fastapi.security import OAuth2PasswordBearer
+
+from syncmaster.backend.dependencies import Stub
+from syncmaster.backend.providers.auth import AuthProvider
+from syncmaster.db.models import User
+
+oauth_schema = OAuth2PasswordBearer(tokenUrl="v1/auth/token", auto_error=False)
+
+
+def get_user(
+    is_active: bool = False,
+    is_superuser: bool = False,
+) -> Callable[[Request, AuthProvider, str], Coroutine[Any, Any, User]]:
+    async def wrapper(
+        request: Request,
+        auth_provider: Annotated[AuthProvider, Depends(Stub(AuthProvider))],
+        access_token: Annotated[str, Depends(oauth_schema)],
+    ) -> User:
+        # keycloak provider patches session and store access_token in cookie,
+        # when dummy auth stores it in "Authorization" header
```

Comment:
If KeycloakAuthProvider does not have such a header, then it should be `str | None` with default value